### PR TITLE
fixes #5865 feat(reporting): Add reportlog frontend 

### LINKED ISSUE #5865: [Reporting]: Reports Frontend
We need a frontend where reportlogs are displayed, buttons to export reportlogs in csv form and in addition to export stats.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1456)

### PR DESCRIPTION
**Pull request by tiftran**
Because:
* We would like to have reportlog and stats displayed in the frontend for consumption

This commit:
* add a frontend page /reporting
* adds csv export functionality to reportlogs and stats

### CODE REVIEW

**Review by jaredlockhart — COMMENTED** [review]
?

Screen Shot 2021-07-27 at 2 37 59 PM

**Comment by jaredlockhart** [comment]
Also probably wanna move the built assets path: 

`app/experimenter/assets/`

to something like 

`app/experimenter/reporting-ui/assets/` 

and add it to .gitignore/.dockerignore

**Comment by jaredlockhart** [comment]
Maybe you could even move `reporting-ui` into `app/experimenter/reporting/`?

**Comment by jaredlockhart** [comment]
Looking good so far! Couple thoughts

* move the filters into that little subheader
* capitalize your column titles
* Add a 'no results returned' message if it comes back empty

File: Makefile [hunk 1]
```diff
@@ -25,8 +25,13 @@ ESLINT_FIX_CORE = yarn workspace @experimenter/core lint-fix
 ESLINT_NIMBUS_UI = yarn workspace @experimenter/nimbus-ui lint
 ESLINT_FIX_NIMBUS_UI = yarn workspace @experimenter/nimbus-ui lint-fix
 TYPECHECK_NIMBUS_UI = yarn workspace @experimenter/nimbus-ui lint:tsc
+ESLINT_REPORTING = yarn workspace @experimenter/reporting lint:eslint --max-warnings=1
+ESLINT_FIX_REPORTING = yarn workspace @experimenter/reporting lint:eslint --fix --max-warnings=1
+TYPECHECK_REPORTING = yarn workspace @experimenter/reporting lint:tsc
```
Comment (jaredlockhart) [inline Makefile:30]: Probably also wanna add the estlint, tsc, and test commands to `make check`?

File: app/experimenter/experiments/views.py [hunk 2]
```diff
@@ -246,3 +246,7 @@ def view_fn(request, exception):
             return response
 
         return view_fn
+
+
+class ExperimentReportView(TemplateView):
+    template_name = "experiments/reporting.html"
```
Comment (jaredlockhart) [inline app/experimenter/experiments/views.py:252]: I think you should move this template into `app/experimenter/reporting/templates`?

File: app/experimenter/reporting/reporting-ui/__tests__/App.test.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import "@testing-library/jest-dom/extend-expect";
```
Comment (jaredlockhart) [inline app/experimenter/reporting/reporting-ui/__tests__/App.test.tsx:1]: Very nice to have the frontend tests 👍

File: app/experimenter/reporting/reporting-ui/components/App.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+const App: React.FC = () => {
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [data, setData] = useState([]);
+  const [statistics, setStatistics] = useState({});
+  const [headers, setHeaders] = useState([]);
+  const onSubmit = async (): Promise<void> => {
+    const response = await fetch(
+      `${window.location.origin}/api/v7/${startDate}/${endDate}`,
```
Comment (jaredlockhart) [inline app/experimenter/reporting/reporting-ui/components/App.tsx:13]: No trailing slash here?? 😱

File: docker-compose.yml [hunk 5]
```diff
@@ -16,6 +16,7 @@ services:
       - ./app:/app
       - /app/experimenter/legacy-ui/core/.cache/
       - /app/experimenter/legacy-ui/core/node_modules/
+      - /app/experimenter/reporting/reporting-ui/node_modules/
```
Comment (jaredlockhart) [inline docker-compose.yml:19]: Make sure to add this to every `app:dev` container below so they all have the same list of volumes

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Awesome work tif, looks great and works great.  Couple little cleanupy things but nothing major.  Couple ideas about the UI:

Screenshot 2021-08-09 at 16-44-29 Mozilla Experimenter
* The `Experiment Reporting` title is super big, maybe look at the other templates to see what header size we use on those?
* Before you had it all in the page body and the subheader was empty, and now you've added both the dates/buttons AND the table itself to the subheader, can you see if you can put the dates/buttons in the subheader but the table in the body?  You might have to move the react root up a block level and then render the subheader and body divs inside your react component.  If it's too much hassle don't worry about it.

File: app/experimenter/experiments/views.py [hunk 6]
```diff
@@ -246,3 +246,7 @@ def view_fn(request, exception):
             return response
 
         return view_fn
+
+
+class ExperimentReportView(TemplateView):
+    template_name = "reporting.html"
```
Comment (tiftran) [inline app/experimenter/experiments/views.py:252]: this is still a little wonky, i tried reporting/reporting.html and reporting-ui/reporting.html and the template paths it made to find the template didn't work out.
did I do something wrong in settings.py?

**Comment by jaredlockhart** [comment]
I think it's fetching the wrong API path?

Screen Shot 2021-08-11 at 11 23 52 AM

File: app/experimenter/urls.py [hunk 7]
```diff
@@ -5,6 +5,7 @@
 
 from experimenter.experiments.views import (
     ExperimentListView,
+    ExperimentReportView,
```
Comment (jaredlockhart) [inline app/experimenter/urls.py:8]: Stick this in `reporting/views.py`

File: app/experimenter/settings.py [hunk 8]
```diff
@@ -104,6 +104,7 @@
         "DIRS": [
             os.path.join(BASE_DIR, "legacy-ui", "templates"),
             os.path.join(BASE_DIR, "nimbus-ui", "templates"),
+            os.path.join(BASE_DIR, "reporting", "reporting-ui", "templates"),
```
Comment (jaredlockhart) [inline app/experimenter/settings.py:107]: This looks right to me 👍

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Yeah just move everything that's particular to reporting into the `reporting/` app

**Review by jaredlockhart — APPROVED** [review]
Awesome all looks right to me \o/ 🎉

**Comment by jaredlockhart** [comment]
Oh and do you wanna add a link to it on the legacy landing page or something?

**Comment by jaredlockhart** [comment]
Oh and we still have the old export as CSV link.  Are we gonna get rid of that?

**Pull request merged**